x/programs/runtime/invoke_program.go

```diff
@@ -25,19 +28,21 @@ const (
 )
 
 type InvokeModule struct {
-	mu      state.Mutable
-	meter   Meter
-	storage Storage
-	log     logging.Logger
+	mu            state.Mutable
+	meter         Meter
+	log           logging.Logger
+	db            database.Database
+	callerAddress ed25519.PublicKey // the callerAddress of the new runtime during invoke.
 }
 
 // NewInvokeModule returns a new program invoke host module which can perform program to program calls.
-func NewInvokeModule(log logging.Logger, mu state.Mutable, meter Meter, storage Storage) *InvokeModule {
+func NewInvokeModule(log logging.Logger, mu state.Mutable, meter Meter, db database.Database, callerAddress ed25519.PublicKey) *InvokeModule {
 	return &InvokeModule{
-		mu:      mu,
-		meter:   meter,
-		storage: storage,
-		log:     log,
+		mu:            mu,
+		meter:         meter,
+		log:           log,
+		db:            db,
+		callerAddress: callerAddress,
```

Comment:
`caller address` isn't being used for anything now. I see the vision for this, but for now I say we remove, and come back to this change to a separate pr. Will create an issue.